src/Apps/Artist/Routes/Consign/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box } from "@artsy/palette"
+import { Match, Router } from "found"
+import React, { useEffect } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+
+import { Consign_artist } from "__generated__/Consign_artist.graphql"
+import { routes_ArtistConsignQueryResponse } from "__generated__/routes_ArtistConsignQuery.graphql"
+
+import { useSystemContext } from "Artsy"
+import { userIsAdmin } from "Utils/user"
+import { ArtistConsignFAQ } from "./Components/ArtistConsignFAQ"
+import { ArtistConsignHeader } from "./Components/ArtistConsignHeader"
+import { ArtistConsignHowtoSell } from "./Components/ArtistConsignHowToSell"
+import { ArtistConsignMarketTrends } from "./Components/ArtistConsignMarketTrends"
+import { ArtistConsignPageViews } from "./Components/ArtistConsignPageViews"
+import { ArtistConsignRecentlySold } from "./Components/ArtistConsignRecentlySold"
+import { ArtistConsignSellArt } from "./Components/ArtistConsignSellArt"
+
+import { getConsignmentData } from "./Utils/getConsignmentData"
+
+interface ConsignRouteProps {
+  artist: Consign_artist
+  artworksByInternalID: routes_ArtistConsignQueryResponse["artworksByInternalID"]
+  match: Match
+  router: Router
+}
+
+const ConsignRoute: React.FC<ConsignRouteProps> = props => {
+  const { user } = useSystemContext()
+  const pathname = props.match.location.pathname.replace("/consign", "")
+  const artistConsignment = getConsignmentData(pathname)
+
+  // FIXME: Ungate admin-only feature when ready to launch
+  const isAuthorizedToView = userIsAdmin(user) && artistConsignment
+
+  // Redirect back to artist overview if artist not found within hand-picked data
+  useEffect(() => {
```

Comment:
Can you help me understand why the redirect is happening inside a hook? Is there something special about this context that makes the redirect work? Will the page flash some amount of the consign markup and then redirect when this hook fires?

Reply:
There's no reason -- and you're right! there's a small flash. I'll guard it behind an `if` and return null.

Reply:
ahh i know why -- its because we do a server-side render pass, where this redirect isn't available (requires window.history). Will update and move to router per your suggestion when we're ready to launch and don't need to worry about guarding against `admin`.